src/main/java/uc/doc/downloaddoc/DownloadDocInteractor.java

```diff
@@ -24,14 +24,15 @@ public DownloadDocInteractor(
     public DownloadDocResponseModel downloadDoc(DownloadDocRequestModel model) {
         User user = stateTracker.getCurrentUser();
         String documentId = model.getDocumentId();
-        String downloadPath;
+        String downloadPath = model.getDownloadPath();
 
         if (!checkIfDocumentDownloaded(documentId)) {
-            downloadPath = getDownloadedDocumentPath(documentId);
             DownloadDocFileAccessRequestModel downloadDocFileAccessRequestModel = new DownloadDocFileAccessRequestModel(documentId, user.getId(), downloadPath);
             if (downloadDocFileAccessGateway.downloadDoc(downloadDocFileAccessRequestModel)) {
                 updateDownloadedDocuments(documentId, downloadPath);
             }
+        } else {
+            downloadPath = getDownloadedDocumentPath(documentId);
```

Comment:
@HarveyRDonnelly is this alright?

Reply:
Yes this looks better but I am still concerned about passing the local path via the request model - this is possible but will complicate the code significantly in the interface adapters layer/views. In my opinion, the simplest way to do this is for the use case to be called anytime we want the local path for a test/solution doc. The use case always returns a local path for the PDF, and downloads the file if it has not been downloaded already.